### ISSUE TITLE
explicit_counter_loop false-positive when the counter is used after the loop

### ISSUE THREAD

**Issue by Ms2ger** [Manishearth/rust-clippy#373]
```rust
                let mut character_count = 0;
                for (character_index, _) in text.char_indices() {
                    if character_index == insertion_point.index {
                        return Some(CharIndex(character_count))
                    }
                    character_count += 1
                }
                return Some(CharIndex(character_count))
```

**Comment by fhartwig** [comment]
I think this is the same as #384 and should be fixed by https://github.com/Manishearth/rust-clippy/pull/473

**Comment by Manishearth** [comment]
https://github.com/Manishearth/rust-clippy/pull/473
